klippa/src/parsing_util.rs

```diff
@@ -86,6 +87,48 @@ pub fn parse_drop_tables(input_str: &str) -> Result<IntSet<Tag>, SubsetError> {
     Ok(result)
 }
 
+//parse input name_IDs string, which is a comma/whitespace-separated list of nameIDs that will be retained
+pub fn parse_name_ids(input_str: &str) -> Result<IntSet<NameId>, SubsetError> {
+    let mut result = IntSet::empty();
+
+    if input_str == "*" {
+        let out = IntSet::<NameId>::all();
+        return Ok(out);
+    }
+
+    for name_id in input_str.split(|c| c == ',' || c == ' ') {
+        if name_id.is_empty() {
+            continue;
+        }
+        let name_id = name_id
+            .parse::<u16>()
+            .map_err(|_| SubsetError::InvalidId(name_id.to_owned()))?;
+        result.insert(NameId::from(name_id));
+    }
+    Ok(result)
+}
```

Comment:
IMHO it would be more idiomatic to use iterators and not bind mut result at all. E.g.

```rust
pub fn parse_name_ids(input_str: &str) -> Result<IntSet<NameId>, SubsetError> {
    if input_str.trim() == "*" {
        return Ok(IntSet::all());
    }
    input_str.split(|c| c == ',' || c == ' ')
        .filter(|raw| !raw.is_empty())
        .map(|raw| raw.parse::<u16>().map(NameId::from).map_err(|_| SubsetError::InvalidId(raw.to_owned())))
        .collect()
}
```

Reply:
Thanks! learnt new rust style:)